wordpress-mobile/WordPress-Android
# [Analytics] Fix for editor_post_published event that is bumped multiple times on save

Fixes #10098 by making sure to bump `editor_post_published` the first time the post is Published on the remote site, and not each time a draft post is saved or updated.

Test 1:
- Start writing a Draft
- Tap on the Save menu
- Re-open and publish it
(there should be only one `editor_post_published` in Logcat)

Test 2:
- Start writing a Draft
- Tap on the back arrow
- Re-open and publish it
(there should be only one `editor_post_published` in Logcat)

Test 3:
- Start writing a Draft
- Tap on the back arrow
- Re-open and just update it
(there should be NO `editor_post_published` in Logcat)


Note: This PR is not intended to fix all the issue with the `editor_post_published` event, but rather to stop the double counting of it, and make analytics go back to "normal" values pre version 12.2

Note 2: There are for sure other flows in the app that publish the post, but doesn't trigger `editor_post_published` event,  ie: moving a post from Private to Published, but those are a vast minority of the normal flows. Also tapping on the notification Retry actions cannot bump the stats fine, but this is another problem.

Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
